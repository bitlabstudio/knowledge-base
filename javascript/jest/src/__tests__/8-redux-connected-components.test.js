import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import TestComponent from '../testedFiles/8-redux-connected-component'

describe('TestComponent', () => {
  const appState = {
    teststore: {
      title: 'My Title',
    },
  }
  let store

  beforeEach(() => {
    store = configureStore()(appState)
  })

  it('renders the correct values', () => {
    const component = renderer.create(
      <Provider store={store}>
        <TestComponent />
      </Provider>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
