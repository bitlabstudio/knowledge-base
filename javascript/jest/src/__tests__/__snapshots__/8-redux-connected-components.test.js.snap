// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent renders the correct values 1`] = `
<h1>
  My Title
</h1>
`;
